File: gpt-3.5-ft-large/template_context_full/declare_directive.cpp
#include "acc_testsuite.h"
#ifndef T1
//T1:declare directive,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    //#pragma acc declare create(a)
    int a[10];

    for (int x = 0; x < 10; ++x){
        a[x] = 0;
    }

    #pragma acc parallel
    {
        #pragma acc loop
        for (int x = 0; x < 10; ++x){
            a[x] = 1;
        }
    }

    for (int x = 0; x < 10; ++x){
        if (a[x] != 1){
            err = 1;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-ft-large/template_context_full/declare_directive.f90
#ifndef T1
!T1:declare directive,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
	INTENT(IN) :: errors
        
        IF (acc_get_device_type() == acc_device_none) THEN
          RETURN
        END IF

        !$acc declare copyin(errors)
        !$acc parallel
          errors = 0
        !$acc end parallel
        IF (errors .ne. 0) THEN
          WRITE(*, *) 'test1:'
          WRITE(*, *) 'Failed to set errors to zero.'
          test1 = .TRUE.
          RETURN
        END IF

        test1 = .FALSE.
        RETURN
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT(failcode)
      END PROGRAM